safekeeper/src/timeline.rs

```diff
@@ -432,11 +435,23 @@ impl Timeline {
                 shared_state.sk.inmem.remote_consistent_lsn,
             )),
             peer_horizon_lsn: Some(shared_state.sk.inmem.peer_horizon_lsn),
-        }
+            wal_stream_connection_string: shared_state
+                .pageserver_connstr
+                .as_deref()
+                .map(|pageserver_connstr| {
```

Comment:
Why do we need pageserver_connstr here? I thought that it's used only by callmemaybe logic, when sk connects to ps and asks it to start replication.

It looks like you're preparing for etcd integration in the pageserver, and we are going away from using callmemaybe. Then you just need to advertise sk callback address, so any pageserver can connect to this safekeeper when it needs to. Am I missing something?

Reply:
I don't think you miss anything, I just follow the path of least resistance and reuse the connection string that was propagated by safekeepers to pageservers before: this way, I have to write no new code to connect and stream the wal.

That connection string was constructed from the `pageserver_connstr` and the `callmemaybe_connstr` logic in the old code, I've removed this in the "remove callmemaybe" request, but left it here.

Reply:
While callmemaybe is still there, we could leave that code in place but publish safekeeper address in SkTimelineInfo already cleanly, without pageserver_connstr appended.

Reply:
Sure, this way we can add the connstr as a separate field later, looks even better.